### DISCUSSION TITLE
How to update search params from input field

### DISCUSSION THREAD

**Discussion by pineshmenat** [remix-run/react-router#8646] [Q&A] [answered]
How to map query string to input field and update it realtime if input field is updated in react router v6?

**Comment by ondrejvelisek** [comment, marked as answer]
Something like this does not work? (not tested)

```
import { useSearchParams } from "react-router-dom";

function App() {
  const [searchParams, setSearchParams] = useSearchParams();

  function onChange(event) {
    setSearchParams(event.target.value);
  }

  return (
      <input value={searchParams} onChange={onChange}/>
  );
}
```

**Reply by pineshmenat** [reply]
Thank you @ondrejvelisek for your help. I also found one good [example](https://reactrouter.com/docs/en/v6/getting-started/tutorial#search-params) in docs
